packages/notebook/src/common/notebook-common.ts

```diff
@@ -298,6 +299,30 @@ export namespace CellUri {
         };
     }
 
+    export function generateCellOutputUri(notebook: URI, outputId?: string): URI {
+        return notebook
+            .withScheme(ouputUriScheme)
+            .withQuery(`op${outputId ?? ''},${notebook.scheme !== 'file' ? notebook.scheme : ''}`);
+    };
+
+    export function parseCellOutputUri(uri: URI): { notebook: URI; outputId?: string } | undefined {
+        if (uri.scheme !== ouputUriScheme) {
+            return;
+        }
+
+        const match = /^op([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})?\,(.*)$/i.exec(uri.query);
+        if (!match) {
+            return undefined;
+        }
+
+        const outputId = (match[1] && match[1] !== '') ? match[1] : undefined;
```

Comment:
Minor: Empty strings are falsy in JavaScript, so we don't need to explicitly check for it.
```suggestion
        const outputId = match[1] || undefined;
```